Backpack/src/main/java/net/skyscanner/backpack/calendar/BpkCalendar.kt

```diff
@@ -2,40 +2,47 @@ package net.skyscanner.backpack.calendar
 
 import android.content.Context
 import android.util.AttributeSet
-import android.widget.LinearLayout
+import android.view.View
+import androidx.constraintlayout.widget.ConstraintLayout
+import net.skyscanner.backpack.R
 import net.skyscanner.backpack.calendar.presenter.BpkCalendarController
 import net.skyscanner.backpack.calendar.view.CalendarView
+import net.skyscanner.backpack.calendar.view.OnYearChangedListener
 import net.skyscanner.backpack.calendar.view.WeekdayHeaderView
+import net.skyscanner.backpack.calendar.view.YearPillView
+import java.util.Calendar
 
 open class BpkCalendar @JvmOverloads constructor(
   context: Context,
   attrs: AttributeSet? = null,
   defStyle: Int = 0
-) : LinearLayout(context, attrs, defStyle) {
+) : ConstraintLayout(context, attrs, defStyle), OnYearChangedListener {
 
   private val calendarView: CalendarView by lazy {
-    CalendarView(context, attrs).also { calendarView ->
-      LinearLayout.LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.MATCH_PARENT)
-        .also { calendarView.layoutParams = it }
-    }
+    findViewById<CalendarView>(R.id.calendar_view)
   }
 
   private val weekdayHeaderView: WeekdayHeaderView by lazy {
-    WeekdayHeaderView(context, attrs)
+    findViewById<WeekdayHeaderView>(R.id.weekday_header_view)
   }
 
-  init {
-    orientation = LinearLayout.VERTICAL
-    layoutParams = LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.MATCH_PARENT)
+  private val yearPillView: YearPillView by lazy {
+    findViewById<YearPillView>(R.id.year_pill_view)
+  }
 
-    addView(weekdayHeaderView)
-    addView(LinearLayout(context).also {
-      it.addView(calendarView)
-    })
+  init {
+    inflate(context, R.layout.view_bpk_calendar, this)
   }
 
   fun setController(controller: BpkCalendarController) {
     weekdayHeaderView.initializeWithLocale(controller.locale)
     calendarView.controller = controller
+    calendarView.listener = this
+  }
+
+  override fun onYearChanged(year: Int) {
+    val currentYear = Calendar.getInstance().get(Calendar.YEAR)
+    yearPillView.visibility = if (currentYear == year) View.GONE else View.VISIBLE
+    yearPillView.postDelayed({ yearPillView.text = year.toString() }, 0)
```

Comment:
@amolgupta @k0nserv I couldn't find any better way to get the text rendered the first time BpkCalendar loads. If I just simply set the text property, the text will not get rendered (all I see is a blank black pill) until the first user interaction happened (user scrolls or taps a date).

Reply:
That could happen if the `onYearChanged` gets called from a non-main thread. Can you think of any way reason why that would happen?

Reply:
This was my first thought also but I checked and it gets called from the main thread.

Reply:
Sure, I don't have a hunch right now what else to do, but we can revisit this in a later PR.